src/containers/base-class/index.js

```diff
@@ -0,0 +1,185 @@
+    this.setState({
+      isModalOpen: true,
+    });
+  };
+
+  requestPayment = (args) => {
+    const amountInCents = parseInt(args.amount, 10);
+
+    if (!amountInCents) {
+      console.log("Invalid amount");
+    } else {
+      this.setState(
+        {
+          isModalOpen: true,
+          currentScreen: "loading-screen",
+        },
+        () => {
+          this.requestPaymentHandler(amountInCents);
+        }
+      );
+    }
+  };
+
+  openWidget = (attributes) => {
+    this.setState({
+      ...attributes,
+      isModalOpen: true,
+    });
+  };
+
+  handleDonateClick = () => {
+    this.setState({
+      currentScreen: "donate-screen",
+    });
+  };
+
+  paymentRequestRenderer = (invoice) => {
+    this.setState({
+      invoiceDetails: invoice,
+      fetchingInvoiceState: false,
+      currentScreen: "payment-screen",
+    });
+  };
+
+  requestPaymentHandler = (amountInCents) => {
+    this.setState({
+      fetchingInvoiceState: true,
+    });
+
+    this.invoiceService
+      .requestPayment({ amount: amountInCents })
+      .then((response) => {
+        if (response.settled) {
+          this.setState({
+            invoiceDetails: null,
+            currentScreen: "thankyou-screen",
+          });
+        }
+      });
+  };
+
+  handleSubscribeClick = (value = "") => {
+    console.log("entered email: ", value);
+  };
+
+  renderModalContent = () => {
+    switch (this.state.currentScreen) {
+      case "welcome-screen":
+        return (
+          <WelcomeScreen
+            message={this.state.welcomeMessage}
+            onDonateClick={this.handleDonateClick}
+          />
+        );
+      case "donate-screen":
+        return (
+          <DonateScreen
+            currency={this.state.currency}
+            paymentOptions={this.state.paymentOptions}
+            isLoading={this.state.fetchingInvoiceState}
+            onNextClick={this.requestPaymentHandler}
+            customAmount={this.state.customAmount}
+          />
+        );
+      case "payment-screen":
+        return <PaymentScreen {...this.state.invoiceDetails} />;
+      case "thankyou-screen":
+        return (
+          <ThankyouScreen
+            thankyouMessage={this.state.thankyouMessage}
+            onSubscribeClick={this.handleSubscribeClick}
+            enableEmailSubscription={this.state.enableEmailSubscription}
+          />
+        );
+      case "loading-screen":
+        return (
+          <LoadingScreen height={40} width={40} color="var(--primary-color)" />
+        );
+      default:
+        null;
+    }
+  };
+}
+
+WidgetBase.defaultProps = {
+  actionLabel: "Donate",
+  imageSrc: "",
+  currency: "EUR",
+  title: "",
+  logo: "",
+  customAmount: true,
+  paymentOptions: { "1€": 1, "2€": 2, "5€": 5, "10€": 10 },
+  screenName: "loading-screen",
+  enableEmailSubscription: false,
+  apiBaseUrl: DEFAULT_API_BASE_URL,
+  apiToken: "",
+  welcomeMessage: "",
+  thankyouMessage: "Thank you!",
+  isModalOpen: false,
```

Comment:
I'd move those specific props to the individual component, they are unrelated to the base and very specific to the modal version. 
or what are your thoughts? 

(see also my PR)